designer/client/component-create.js

```diff
@@ -6,6 +6,13 @@ import ComponentTypes from 'digital-form-builder-engine/src/component-types'
 class ComponentCreate extends React.Component {
   state = {}
 
+  componentDidMount () {
+    const { data, onInit } = this.props
+    data.getId().then(id => {
```

Comment:
This might help you with the test, maybe this should be in the constructor? instead of calling this.setState, assign state with this.state = { }, then you can call onInit wherever it's most suitable.

Reply:
Sadly I don't think it helps, unless I'm misunderstanding something. I can't do await in the constructor because the constructor cannot be async. So I'm still left with data.getId().then(....) which, by its nature, cannot be tested without having a callback to let me know it's finished.